### PR TITLE
moved password and mail validation to backend

### PR DESCRIPTION
**Pull request by DominykasSip**

### CODE REVIEW

File: wasted-app/RestaurantRegistrationControl.cs [hunk 1]
```diff
@@ -60,29 +61,16 @@ private void RegisterButton_Click(object sender, EventArgs e)
...
-                        controller.RestaurantService.Register(creds, new Restaurant("todo", restaurantNameTextBox.Text, new Coords(Convert.ToDecimal(latitudeTextBox.Text), Convert.ToDecimal(longitudeTextBox.Text)), new Credentials()));
-                        MessageBox.Show("Registered successfully");
-                        GoBack();
-                        ResetTextBoxes();
-                    }
-                    else
-                    {
-                        passwordError.Text = credentialError;
-                    }
+                    controller.RestaurantService.Register(new Credentials(mailTextBox.Text, passwordTextBox.Text), new Restaurant("todo", restaurantNameTextBox.Text, new Coords(Convert.ToDecimal(latitudeTextBox.Text), Convert.ToDecimal(longitudeTextBox.Text)), new Credentials()));
```
Comment (DonatasPuc) [inline wasted-app/RestaurantRegistrationControl.cs:66]: This is quite unreadable in my opinion. Maybe extract arguments to variables or at least put arguments on different lines.

Comment (audrius-savickas) [reply wasted-app/RestaurantRegistrationControl.cs:66]: I'd say better to move into separate variables

Comment (DominykasSip) [reply wasted-app/RestaurantRegistrationControl.cs]: Fixed it.

File: wasted-app/RestaurantRegistrationControl.cs [hunk 2]
```diff
@@ -62,27 +63,22 @@ private void RegisterButton_Click(object sender, EventArgs e)
                 ServicesController controller = ServicesController.Instance;
                 var mail = mailTextBox.Text;
                 var password = passwordTextBox.Text;
-                var restaurant = controller.RestaurantService.GetByMail(new Mail(mail));
-                if (restaurant == null)
+                var restaurantName = restaurantNameTextBox.Text;
+                var latitude = Convert.ToDecimal(latitudeTextBox.Text);
+                var longitude = Convert.ToDecimal(longitudeTextBox.Text);
```
Comment (audrius-savickas) [inline wasted-app/RestaurantRegistrationControl.cs:68]: program probably crashes if input is empty or non-numeric, right?

Comment (DominykasSip) [reply wasted-app/RestaurantRegistrationControl.cs:68]: yea but we decided to leave it for now, right?

Comment (DominykasSip) [reply wasted-app/RestaurantRegistrationControl.cs:68]: I will check maybe there is a quick fix for this in textbox parameters

Comment (audrius-savickas) [reply wasted-app/RestaurantRegistrationControl.cs:68]: nope, by default textbox can't do any validation. you could check my latest closed pr to see how I did it, to only allow numbers to be put in  (#67 ), except it still crashes when it's empty.

Comment (DominykasSip) [reply wasted-app/RestaurantRegistrationControl.cs:68]: shouldn't crash now

**Review by audrius-savickas — APPROVED** [review]

**Review by DonatasPuc — APPROVED** [review]

**Pull request merged**